src/community/features-templating/features-templating-core/src/main/java/org/geoserver/featurestemplating/expressions/TemplateCQLManager.java

```diff
@@ -284,11 +284,6 @@ public static String removeQuotes(String cqlFilter) {
             char curr = cqlFilter.charAt(i);
             if (curr != '\"') {
                 strBuilder.append(curr);
-            } else {
```

Comment:
The filter needs to be passed to the OGCAPI controller as a string. The toCQL methods wraps a property name into "\"" and that method remove the quotes. The else statement was meant to avoid replacing them in the case of a liter, but literals are actually encoded using ' ' from the toCQL method, so that the piece of code in question was useless and actually potentially causing malfunctioning of the backwards mapping.